merlin/core/dispatch.py

```diff
@@ -269,13 +270,13 @@ def series_has_nulls(s):
         return s.has_nulls
 
 
-def list_val_dtype(ser: SeriesType) -> np.dtype:
```

Comment:
Replacing `SeriesType` and `DataframeType` (an ongoing pain with MyPy, which does not like dynamically defined types) with the protocols here doesn't functionally change much here, since the actual types don't prevent you from passing the wrong types into the function. The protocols (e.g. `SeriesLike`) can be checked at run-time via `isinstance(obj, protocol)` checks. Not sure if there's a way to enforce them automatically at runtime though.